### DISCUSSION TITLE
Loss function

### DISCUSSION THREAD

**Discussion by LshanH** [darioizzo/geodesyNets#109] [Q&A]
- Question 1: 
How to select the batch_size in during training when making mask? Turn to the loss function, if involving batch_size=m, the loss function should be set as this way?

$$loss=\frac{1}{\mathrm{~m}} \sum_{\mathrm{j}=1}^{\mathrm{m}}\left(\frac{1}{\mathrm{n}} \sum_{\mathrm{i}=1}^{\mathrm{n}}\left|\mathrm{y}_{\mathrm{i}}-\mathrm{k} * \hat{\mathrm{y}}_{\mathrm{i}}\right|\right)
$$

- Question 2: 
In the drawing module (def plot_model_rejection ()), we first generate candidate points with uniform distribution. Should we judge the candidate points whether within the model or not? Then we used the candidate density and the mask together, but here what is the purpose of the mask here?
`_rho_candidates = model(nn_inputs).detach() * c`
`
mask = rho_candidates > (torch.rand(batch_size , 1) + crop_p)_
`
What is the role of c in this case? And then plus a density minimum value (cross _ p), how to get this cross_p value?

- Question 3: 
In the drawing module (def plot_model_rejection), it uses a judgment that the candidate point is empty or not,
`_if len (candidates) == 0:`
`print ("All points rejected!Plot is empty, try cropping less?")`
`return_`

- Question 4: 
In the process of differentiation training, because the density can be a negative value, the mask may lead to the model rejection in the drawing part? My current results for Differential Training are shown below
![image](https://user-images.githubusercontent.com/24770360/221062355-67b9be23-415f-4acd-a72e-d9d8dc829734.png)
![image](https://user-images.githubusercontent.com/24770360/221062366-dd2a3692-5df6-456f-b6a0-4b07115a1c30.png)

**Comment by darioizzo** [comment]
I am trying to answer your question, but it is difficult as I am not sure I understand them.
Q1. -> for the batch_size you can refer to our notebooks: 
```
# Use 100 for a quick training. 1000  was used to produce most of the paper results
batch_size = 1000
```
Q2. "What is the role of c in this case?" -> The asteroid density is c times the output of the network. c is what we also call k in the paper.
Q3. No question is asked
Q4. I dot understand your question. Your plot shows you set up something wrong.
